libweasyl/alembic/versions/57171ee9e989_make_birthdate_optional.py

```diff
@@ -0,0 +1,34 @@
+"""Make birthdate optional
+
+Revision ID: 57171ee9e989
+Revises: 9a41d4fa38ba
+Create Date: 2023-12-22 00:15:53.754117
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '57171ee9e989'
+down_revision = '9a41d4fa38ba'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.alter_column('logincreate', 'birthday',
+               existing_type=sa.INTEGER(),
+               nullable=True)
+    op.add_column('userinfo', sa.Column('asserted_adult', sa.Boolean(), server_default='f', nullable=False))
+    op.alter_column('userinfo', 'birthday',
+               existing_type=sa.INTEGER(),
+               nullable=True)
+
+
+def downgrade():
+    op.alter_column('userinfo', 'birthday',
+               existing_type=sa.INTEGER(),
+               nullable=False)
```

Comment:
Observation: Doesn't this cause an issue? If the upgrade step permits a null value in `userinfo`, wouldn't this constraint then fail for any users who have signed up after the upgrade is run in the event a downgrade is run after?

Reply:
Yes, it will fail. If we want to downgrade after people have signed up with this deployed, we’ll need to decide what we want to do with those accounts. But I hope we won’t want to downgrade :D